addon/components/o-s-s/chip.ts

```diff
@@ -0,0 +1,66 @@
+import Component from '@glimmer/component';
+import { assert } from '@ember/debug';
+import { action } from '@ember/object';
+
+export type SkinType = 'default' | 'primary' | 'success' | 'danger';
+
+type SkinDefType = {
+  [key in SkinType]: string;
+};
+
+export const SkinDefinition: SkinDefType = {
+  default: 'default',
+  primary: 'primary',
+  success: 'success',
+  danger: 'danger'
+};
+
+const BASE_CLASS_CHIP: string = 'upf-chip';
+interface OSSChipArgs {
+  skin?: string;
+  label?: string;
+  onClick(crossClick: boolean): void;
+  disabled?: boolean;
+  ellipsisWidth?: number;
```

Comment:
Tech spec defines the last width parameter as "displayMaxWidth" ; linked to my previous comment about not using the `text-ellispis-xx` classes, I think it makes more sense to use this keyword (or something similar) instead of `ellipsisWidth`